src/chains/ethereum/block/src/runtime-block.ts

```diff
@@ -156,21 +171,27 @@ export class RuntimeBlock {
       header.mixHash,
       BUFFER_8_ZERO // nonce
     ];
-    if (this.serializeBaseFeePerGas && header.baseFeePerGas !== undefined) {
-      rawHeader[15] = Quantity.toBuffer(header.baseFeePerGas);
+    const isEip4895 = this._common.isActivatedEIP(4895);
+    // baseFeePerGas was added in London
+    if (header.baseFeePerGas !== undefined) {
+      rawHeader[15] = Quantity.toBuffer(header.baseFeePerGas, false);
+      // withdrawalsRoot was added in Shanghai
+      if (isEip4895) rawHeader[16] = Data.toBuffer(header.withdrawalsRoot);
     }
 
     const { totalDifficulty } = header;
-    const txs: TypedDatabaseTransaction[] = [];
-    const extraTxs: GanacheRawBlockTransactionMetaData[] = [];
-    transactions.forEach(tx => {
-      txs.push(<TypedDatabaseTransaction>tx.raw);
-      extraTxs.push([tx.from.toBuffer(), tx.hash.toBuffer()]);
-    });
-    const { serialized, size } = serialize([
-      rawHeader,
-      txs,
-      [],
+    const txs: BlockRawTransaction[] = Array(transactions.length);
+    const extraTxs: GanacheRawBlockTransactionMetaData[] = Array(
+      transactions.length
+    );
+    for (let i = 0; i < transactions.length; i++) {
+      const tx = transactions[i];
+      txs[i] = tx.raw.length === 9 ? tx.raw : tx.serialized;
+      extraTxs[i] = [tx.from.toBuffer(), tx.hash.toBuffer()];
+    }
+    let rawBlock: EthereumRawBlock;
+    rawBlock = isEip4895 ? [rawHeader, txs, [], []] : [rawHeader, txs, []];
```

Comment:
So this on two lines only because if I put it on one it puts it on lots of lines, and I didn't like it. haha. Not sure I like this either though. So it is up to you, dear reviewer, to decide which way you hate more.

Reply:
I'd vote for this, because (although not _super_ tidy) functionally different (better).
```suggestion
    const rawBlock: EthereumRawBlock = isEip4895
      ? [rawHeader, txs, [], []]
      : [rawHeader, txs, []];
```